back-end/service.py

```diff
@@ -1,172 +1,301 @@
-	os.makedirs(clone_path, exist_ok=True)
+    with open(
+        os.path.join('templates', 'Snakefile.template'),
+            'r+') as template:
 
-	call(['git', 'clone', '--depth=1', repo.urls.clone_url, clone_path])
+        for line in template.readlines():
+            if '#Add-Files' not in line: 
+                lines.append(line)
+            else:
+                input_str = ""
+                for filename in files.values():
+                    input_str += "'{}',".format(filename)
 
-	print("Repo cloned at " + clone_path)
-	return clone_path
+                input_str = input_str[:-1]
+                lines.append("SAMPLES = [{}]".format(input_str))
 
-def clean(path):
-	call(['rm', '-rf', path])
+    template.close()
+
+    with open(os.path.join(repoPath, 'Snakefile'), 'w+') as config: 
+        config.writelines(lines)
+
+    config.close()
+
+    print("Workflow written at " + repoPath + "/Snakefile")
+
+
+def addNotifications(lines, notifications):
+
+    for notif in notifications:
+        if notif['value']:
+            if notif['name'] == "slack":
+                print("notification detected: {}".format("slack"))
+                with open(
+                    os.path.join('templates', 'slack.template'),
+                        'r+') as slack:
+
+                        for notif_lines in slack:
+                            lines.append(notif_lines)
+
+
+def designCIConfig(notifications, backPushfiles, annexFiles, repoPath):
+    lines = []
+
+    with open(
+        os.path.join('templates', 'drone.template'),
+            'r+') as template:
+
+        for line in template.readlines():
+
+            if '#add-backpush-files' in line:
+                if len(backPushfiles) > 0:
+                    input_files = ""
+                    for filename in backPushfiles.values():
+                        input_files += "'{}' ".format(filename)
+                    lines.append(('    - mv {} "$TMPLOC"').format(input_files))
+
+            elif '#copy-backpush-files' in line:
+                if len(backPushfiles) > 0:
+                    input_files = ''
+                    for filename in backPushfiles.values():
+                        input_files += '"$TMPLOC"/"{}" '.format(filename)
+                    lines.append('    - mv {} "$DRONE_BUILD_NUMBER"/'.format(
+                        input_files))
+
+            elif '#get-annexed-files' in line:
+                if len(annexFiles.values()) == 0:
+                    lines.append("    - git annex sync --content")
+                else:
+                    input_files = ""
+                    for filename in annexFiles.values():
+                        input_files += "'{}' ".format(filename)
+                    lines.append('    - git annex get {}'.format(input_files))
+
+            elif '#add-notifications' in line:
+                addNotifications(lines, notifications)
+
+            else:
+                lines.append(line)
+
+    template.close()
```

Comment:
When using the `with` statement, explicitly closing the file afterwards is not necessary, the ContextManager takes care of that.

Reply:
Removed the closing tags with all `with` statements.